java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java

```diff
@@ -289,16 +290,29 @@ private boolean isPublic() {
 
   public boolean isAnnotatedOverride() {
     for (AnnotationTree annotationTree : modifiers.annotations()) {
-      if (annotationTree.annotationType().is(Tree.Kind.IDENTIFIER)) {
-        IdentifierTree identifier = (IdentifierTree) annotationTree.annotationType();
-        if (Override.class.getSimpleName().equals(identifier.name())) {
-          return true;
-        }
+      if (annotationTree.annotationType().is(Tree.Kind.IDENTIFIER) && isOverride((IdentifierTree) annotationTree.annotationType())) {
+        return true;
+      } else if (annotationTree.annotationType().is(Kind.MEMBER_SELECT) && isJavaLangOverride((MemberSelectExpressionTree) annotationTree.annotationType())) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 293. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1871)